lib/box2d/contact_callbacks.dart

```diff
@@ -1,26 +1,27 @@
 import 'package:box2d_flame/box2d.dart';
 import 'package:flame/box2d/box2d_component.dart';
 
-class ContactTypes {
-  final Type type1, type2;
-
-  ContactTypes(this.type1, this.type2);
-
-  bool has(Type type) => type1 == type || type2 == type;
-
-  // If it contains at least one type from the other ContactTypes
-  bool hasOneIn(ContactTypes other) => has(other.type1) || has(other.type2);
-
-  bool equals(ContactTypes other) =>
-      (type1 == other.type1 && type2 == other.type2) ||
-          (type2 == other.type1 && type1 == other.type2);
+class ContactTypes<T1, T2> {
+  // If o1 is, or inherits from, T1 or T2
+  bool has(Object o1) => o1 is T1 || o1 is T2;
+  bool hasOne(Object o1, Object o2) => has(o1) || has(o2);
```

Comment:
I found the `has` confusing here, what about: `is` instead of `has`?

Reply:
`Is` would be confusing since that lingo is already used for types in Dart, and since this checks two objects and not only one. Would `contains` make it less confusing maybe?

Reply:
Idk, contains seems weird to me as well hahahahaha, I think it is ok to leave a `has` then, it was a minor suggestion in the end.

Reply:
It's only used internally anyways. :)